Packages/Tracking Preview/Input Action Hands/LeapHandUpdater.cs

```diff
@@ -0,0 +1,158 @@
+/******************************************************************************
+ * Copyright (C) Ultraleap, Inc. 2011-2022.                                   *
+ *                                                                            *
+ * Use subject to the terms of the Apache License 2.0 available at            *
+ * http://www.apache.org/licenses/LICENSE-2.0, or another agreement           *
+ * between Ultraleap and you, your company or other organization.             *
+ ******************************************************************************/
+
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+using Leap.Unity;
+using Leap;
+using UnityEngine.InputSystem;
+using UnityEngine.InputSystem.Layouts;
+
+namespace Leap.Unity.Preview.InputActions
+{
+    public class LeapHandUpdater : PostProcessProvider
+    {
+        private LeapHandState _leftState = new LeapHandState(), _rightState = new LeapHandState();
+
+        private bool _leftReportedNull = false, _rightReportedNull = false;
+
+        private InputDevice _leftDevice, _rightDevice;
+
+        public override void ProcessFrame(ref Frame inputFrame)
+        {
+            if (Application.isPlaying)
+            {
+                if (LeapHandInput.leftHand != null)
+                {
+                    Hand hand = inputFrame.GetHand(Chirality.Left);
+                    if (hand != null)
+                    {
+                        _leftReportedNull = false;
+                        ConvertLeapHandToState(ref _leftState, hand);
+                        InputSystem.QueueStateEvent(LeapHandInput.leftHand, _leftState);
+                    }
+                    else
+                    {
+                        if (!_leftReportedNull)
+                        {
+                            _leftState.tracked = 0;
+                            InputSystem.QueueStateEvent(LeapHandInput.leftHand, _leftState);
```

Comment:
Do we need to clear any other state info other than the tracked state when the hand is null?

Reply:
I don't know whether we should/shouldn't essentially, 'tracked' should be being used to determine what data is up-to-date, so if it's false, you still may want the most recent data?